website_sale_checkout_skip_payment/controllers/main.py

```diff
@@ -31,15 +25,13 @@ def shop_payment_confirmation(self, **post):
             return super().shop_payment_confirmation(**post)
         order = request.env["sale.order"].sudo().browse(order_id)
         try:
-            order.with_context(mark_so_as_sent=True)._send_order_confirmation_mail()
+            order.with_context(send_email=True, mark_so_as_sent=True).with_user(
+                SUPERUSER_ID
+            ).action_confirm()
```

Comment:
This is done like this for a reason, as pointed some lines below. With overrides like the one `sale_financial_risk` on `action_confirm` returning a wizard when the partner risk is exceeded, the cart wouldn't be closed. So better leave it as it was.